SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/DynamicSubobjectsTest/DynamicSubobjectsTest.cpp

```diff
@@ -52,198 +70,251 @@ void ADynamicSubobjectsTest::PrepareTest()
+	AddStep(TEXT("DynamicSubobjectsTestSetup"), FWorkerDefinition::Server(1), nullptr, [this]() {
+		ASpatialFunctionalTestFlowController* ClientOneFlowController = GetFlowController(ESpatialFunctionalTestWorkerType::Client, 1);
+		APlayerController* PlayerController = Cast<APlayerController>(ClientOneFlowController->GetOwner());
 
-			if (IsValid(PlayerController))
-			{
-				ClientOneSpawnedPawn =
-					GetWorld()->SpawnActor<ATestMovementCharacter>(CharacterSpawnLocation, FRotator::ZeroRotator, FActorSpawnParameters());
-				RegisterAutoDestroyActor(ClientOneSpawnedPawn);
+		if (AssertIsValid(PlayerController, TEXT("PlayerController should be valid")))
+		{
+			ClientOneSpawnedPawn = GetWorld()->SpawnActor<ATestMovementCharacter>(CharacterSpawnLocation, FRotator::ZeroRotator);
+			RegisterAutoDestroyActor(ClientOneSpawnedPawn);
 
-				ClientOneDefaultPawn = PlayerController->GetPawn();
+			ClientOneDefaultPawn = PlayerController->GetPawn();
+			PlayerController->Possess(ClientOneSpawnedPawn);
 
-				PlayerController->Possess(ClientOneSpawnedPawn);
+			FinishStep();
+		}
+	});
+
+	// Step 1 - All workers check if they have one AReplicatedGASTestActor in the world, and set a reference to it.
+	AddStep(
+		TEXT("DynamicSubobjectsTestAllWorkers"), FWorkerDefinition::AllWorkers, nullptr, nullptr,
+		[this](float DeltaTime) {
+			TestActor = GetReplicatedTestActor();
+			FinishStep();
+		},
+		StepTimeLimit);
 
+	// Step 2 - Client 1 checks if it has correctly possessed the TestMovementCharacter.
+	AddStep(
+		TEXT("DynamicSubobjectsTestClientCheckPossesion"), FWorkerDefinition::Client(1), nullptr, nullptr,
+		[this](float DeltaTime) {
+			APawn* PlayerCharacter = GetFlowPawn();
+			if (AssertIsValid(PlayerCharacter, TEXT("PlayerCharacter should be valid")))
+			{
+				RequireTrue(PlayerCharacter == GetFlowPlayerController()->AcknowledgedPawn, TEXT("The client should possess the pawn."));
 				FinishStep();
 			}
+		},
+		StepTimeLimit);
+
+	// Step 3 - The client checks it has the right initial amount of components
+	AddStep(TEXT("DynamicSubobjectsTestClientCheckNumComponents"), FWorkerDefinition::Client(1), nullptr, [this]() {
+		AssertEqual_Int(GetNumComponentsOnTestActor(), 3, TEXT("For this test, DynamicSubobjectTestActor should have 3 components"));
+		FinishStep();
+	});
+
+	// Step 4 - The server adds the new dynamic component
+	AddStep(TEXT("DynamicSubobjectsTestServerAddComponent"), FWorkerDefinition::Server(1), nullptr, [this]() {
+		AssertEqual_Int(GetNumComponentsOnTestActor(), 3, TEXT("For this test, DynamicSubobjectTestActor should have 3 components"));
+
+		// add new dynamic component to test actor
+		USceneComponent* AddedComponent = NewObject<USceneComponent>(TestActor, TEXT("ToRemoveComponent"));
+		AddedComponent->AttachToComponent(TestActor->GetRootComponent(), FAttachmentTransformRules::KeepWorldTransform);
+		AddedComponent->RegisterComponent();
```

Comment:
Does this function not exist anymore?

Reply:
what do you mean?

Reply:
o i guess you're talking about the function i had for this, ye i removed it at some point